### ISSUE TITLE
expectAssertion string match

### ISSUE THREAD

**Issue by MichalBryxi** [workmanw/ember-qunit-assert-helpers#11]
```javascript
// component.js
...
Ember.assert('{{#th-order-by}} must pass value argument: ["foo", "bar"]', myCheck);
...
```

```javascript
// component-test.js
...
assert.expectAssertion(
    () => {
      this.render(hbs`{{th-order-by}}`);
    },
    "Assertion Failed: {{#th-order-by}} must pass value argument: [\"foo\", \"bar\"]"
  );
});
...
```

Produces:

<img width="775" alt="screen shot 2017-06-30 at 11 23 07" src="https://user-images.githubusercontent.com/847473/27731810-89b4e0f0-5d86-11e7-96fc-c4526c2b7aca.png">

- Tests say that `expected` and `actual` are the same and yet the test itself **fails**.
- I guess the problem is that the helper automatically assumes that `matcher` parameter is regular expression. So internally it will let the test assertion fail. Because I'm using characters that are special for regular expressions (hash, bracket, ...). But then when the strings are passed to testing environment, they really match and thus I'm getting the message that everything is ok. 

1. Would be sweet to have clear message to the user that does not produce false positives (actual == expected, yet test fails)
1. Would be awesome to have this explicitly stated in documentation as an example. That the helper expects regular expression.
1. Ban passing anything that is not regular expression into the helper? Or allow simplifies string matching (`===`) when string is passed as matcher?

**Comment by workmanw** [comment]
Hey @MichalBryxi -- Sorry for the delay. 

I originally tried to mimic the functionality of [ember-try](https://github.com/emberjs/ember-dev), but with QUnit 2.0 compatibility in hopes of being a replacement for it at some point.

That said, I can completely see how this is a gun foot. What would you say about doing an `instanceof RegExp` check? Strings could use `String.prototype.includes()`, while regex could do traditional matching. Seems like a win / win?

**Comment by MichalBryxi** [comment]
No worries @workmanw. Happy when this one gets better. 
Your proposal is something I did not propose to make things simple. But frankly I see your solution as the best one. If that is covered in the docs, then I really think it should differentiate between RegExp / String.

**Comment by workmanw** [comment]
Closed by #16.
